tests/libgstc/c/test_libgstc_pipeline_list_signals.c

```diff
@@ -0,0 +1,237 @@
+
+static void
+setup (void)
+{
+  const gchar *address = "";
+  unsigned int port = 0;
+  unsigned long wait_time = 5;
+  int keep_connection_open = 0;
+
+  gstc_client_new (address, port, wait_time, keep_connection_open, &_client);
+}
+
+static void
+teardown (void)
+{
+  gstc_client_free (_client);
+}
+
+/* Mock implementation of a socket */
+typedef struct _GstcSocket
+{
+} GstcSocket;
+
+GstcSocket _socket;
+
+GstcStatus
+gstc_socket_new (const char *address, const unsigned int port,
+    const int keep_connection_open, GstcSocket ** out)
+{
+  *out = &_socket;
+
+  return GSTC_OK;
+}
+
+void
+gstc_socket_free (GstcSocket * socket)
+{
+}
+
+GstcStatus
+gstc_socket_send (GstcSocket * socket, const gchar * request, gchar ** response,
+    const int timeout)
+{
+  *response = malloc (1);
+
+  memcpy (_request, request, strlen (request));
+
+  return GSTC_OK;
+}
+
+GstcStatus
+gstc_json_get_int (const gchar * json, const gchar * name, gint * out)
+{
+  return *out = GSTC_OK;
+}
+
+GstcStatus
+gstc_json_is_null (const gchar * json, const gchar * name, gint * out)
+{
+  *out = 0;
+  return GSTC_OK;
+}
+
+GstcStatus
+gstc_json_get_child_char_array (const char *json, const char *parent_name,
+    const char *array_name, const char *element_name, char **out[],
+    int *array_lenght)
+{
+  gstc_assert_and_ret_val (NULL != json, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != parent_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != array_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != element_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != out, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != array_lenght, GSTC_NULL_ARGUMENT);
+  return GSTC_OK;
+}
+
+GstcStatus
+gstc_json_child_string (const char *json, const char *parent_name,
+    const char *data_name, char **out)
+{
+  gstc_assert_and_ret_val (NULL != json, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != parent_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != data_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != out, GSTC_NULL_ARGUMENT);
+
+  return GSTC_OK;
+}
+
+GST_START_TEST (test_pipeline_list_signals_success)
+{
+  GstcStatus ret;
+  const gchar *pipeline_name = "pipe";
+  const gchar *element_name = "element_name";
+  const gchar *expected = "read /pipelines/pipe/elements/element_name/signals";
+  char **response;
+  int array_lenght;
+
+  ret =
+      gstc_pipeline_list_signals (_client, pipeline_name, element_name,
+      &response, &array_lenght);
+  assert_equals_int (GSTC_OK, ret);
+
+  assert_equals_string (expected, _request);
+}
+
+GST_END_TEST;
+
+GST_START_TEST (test_pipeline_list_signals_null_pipeline)
+{
+  GstcStatus ret;
+  const gchar *pipeline_name = NULL;
+  const gchar *element_name = "element_name";
+  char **response;
+  int array_lenght;
+
+  ret =
+      gstc_pipeline_list_signals (_client, pipeline_name, element_name,
+      &response, &array_lenght);
```

Comment:
response should be freed here, let's do it in case someone takes this as a reference